src/storage/mod.rs

```diff
@@ -390,7 +399,27 @@ mod test {
                 compression: Some(*alg),
             };
             test_roundtrip(std::slice::from_ref(&blob));
-            assert_eq!(decompress(data.as_slice(), *alg).unwrap(), orig.as_bytes());
+            assert_eq!(
+                decompress(data.as_slice(), *alg, std::usize::MAX).unwrap(),
+                orig.as_bytes()
+            );
+        }
+    }
+
+    #[test]
+    fn test_decompression_too_big() {
+        let orginal = &[b'A'; 2000] as &[u8];
+
+        for alg in CompressionAlgorithm::AVAILABLE {
+            let data = compress(orginal, *alg).unwrap();
+
+            // Ensure decompressing a file over the limit returns a SizeLimitReached error.
+            let err = decompress(data.as_slice(), *alg, 1000).unwrap_err();
+            assert!(err
+                .downcast_ref::<std::io::Error>()
+                .and_then(|io| io.get_ref())
+                .and_then(|err| err.downcast_ref::<crate::error::SizeLimitReached>())
+                .is_some());
         }
```

Comment:
Testing edge cases here as well would be good, like undersized and files that are on the buffer limit

Reply:
Good point.